src/EPiServer.SocialAlloy.Web/Social/Repositories/ActivityStreams/PageSubscriptionRepository.cs

```diff
@@ -167,17 +165,21 @@ private IEnumerable<PageSubscription> AdaptSocialSubscription(List<Subscription>
         }
 
         /// <summary>
-        /// Adapt a PageSubscriptionFilter to a SubscriptionFilter
+        /// Adapt a PageSubscriptionFilter to a FilterExpression
         /// </summary>
         /// <param name="filter">The PageSubscriptionFilter </param>
-        /// <returns>The SubscriptionFilter</returns>
-        private SubscriptionFilter AdaptSubscriptionFilter(PageSubscriptionFilter filter)
+        /// <returns>The FilterExpression</returns>
+        private FilterExpression AdaptSubscriptionFilter(PageSubscriptionFilter filter)
         {
-            return new SubscriptionFilter
-            {
-                Subscriber = !string.IsNullOrWhiteSpace(filter.Subscriber) ? Reference.Create(filter.Subscriber) : Reference.Empty,
-                Target = !string.IsNullOrWhiteSpace(filter.Target) ? Reference.Create(filter.Target) : Reference.Empty,
-            };
+            var subscriberFilter = this.subscriptionFilters.Subscriber.EqualTo(
+                !string.IsNullOrWhiteSpace(filter.Subscriber) ? filter.Subscriber : ""
+            );
+
+            var targetFilter = this.subscriptionFilters.Target.EqualTo(
+                !string.IsNullOrWhiteSpace(filter.Target) ? filter.Target : ""
+            );
```

Comment:
for the conditional return to work properly filters must be initiated as:

var filters = new List<ListExpression>();